ModularTegustation/tegu_items/gadgets/unpowered.dm

```diff
@@ -295,3 +295,69 @@
 	if(istype(H) && (H?.mind?.assigned_role == "Clerk"))
 		return TRUE
 	return FALSE
+
+
+
+
+//Tool E.G.O extractor
+/obj/item/tool_extractor
+	name = "Enkephalin Resonance Unit"
+	desc = "A specialized set of tools that allows E.G.O extraction from tool abnormalities."
+	icon = 'icons/obj/storage.dmi'
+	icon_state = "RPED"
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = ITEM_SLOT_BELT
+	var/stored_enkephalin = 0
+	var/maximum_enkephalin = 250
+	var/ego_selection
+	var/ego_array
+
+/obj/item/tool_extractor/examine(mob/user)
+	. = ..()
+	. += "Currently storing [stored_enkephalin]/[maximum_enkephalin] enkephalin."
+
+/obj/item/tool_extractor/attack_obj(obj/O, mob/living/carbon/user)
+	if(user.a_intent == INTENT_HARM)
+		return ..()
+	if(istype(O ,/obj/machinery/computer/extraction_cargo))//console stuff here
+		var/drawn_amount = 50
+		if(stored_enkephalin > 200)
+			drawn_amount = (maximum_enkephalin - stored_enkephalin)//top off without going over the max
+		if(SSlobotomy_corp.available_box < drawn_amount)
+			to_chat(usr, "<span class='warning'>There is not enough enkephalin stored for this operation.</span>")
+			playsound(get_turf(src), 'sound/machines/terminal_prompt_deny.ogg', 50, TRUE)
+			return FALSE
+		if(stored_enkephalin == maximum_enkephalin)
+			to_chat(usr, "<span class='warning'>[src] is at full capacity..</span>")
+			playsound(get_turf(src), 'sound/machines/terminal_prompt_deny.ogg', 50, TRUE)
+			return FALSE
+		stored_enkephalin += drawn_amount
+		SSlobotomy_corp.AdjustAvailableBoxes(-1 * drawn_amount)
+		playsound(get_turf(src), 'sound/machines/terminal_prompt_confirm.ogg', 50, TRUE)
+		to_chat(usr, "Transferred [drawn_amount] enkephalin into [src].")
+		return ..()
+	if(!istype(O, /obj/structure/toolabnormality))//E.G.O stuff below here
+		return ..()
+	var/obj/structure/toolabnormality/P = O
+	ego_selection = input(user, "Which E.G.O will you extract?") as null|anything in P.ego_list
+	if(!ego_selection)
+		return ..()
+	var/datum/ego_datum/D = ego_selection
+	var/enkephalin_cost = initial(D.cost)//cannot call cost from D. causes a runtime
+	var/loot = initial(D.item_path)//cannot call cost from D. causes a runtime
+	switch(enkephalin_cost)
+		if(20 to 34)
+			enkephalin_cost *= 2
+		if(35 to 44)//this excludes WAW rank-ups which usually cost 45 vs 50
+			enkephalin_cost *= 3
+		if(45 to 99)
+			enkephalin_cost *= 4
+		if(100 to INFINITY)//unobtainable
+			enkephalin_cost *= 5
```

Comment:
I think this is too much ramp and really only needs 1.5 for WAW and 2.0 for ALEPH

Reply:
I'm a bit skeptical about this one since it uses cargo supply PE which is relatively inexpensive, but it's worth a try.